### PR TITLE
ci: implement guix build by label request in CI 

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Automated guix builds in CI when specifically requested 

## What was done?
Any PR with the `build-guix` label added will automatically have the Guix build ran and the hashes placed in the CI output to compare against

## How Has This Been Tested?
This PR

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
without label no guix ci
image

**Comment by PastaPastaPasta** [comment]
I will drop this commit a42ef540e4879f21e0bc3970f276fb4ddf51dff3 before we merge this but I don't want to trigger a new build now, so please review first, then I'll drop this commit and merge

**Comment by knst** [comment]
> I think this is ready for review, in a week or so @strophy will help me figure out better caching and better runner usage that is cheaper (as guix isn't amazingly multi-threaded it seems, lots of single threaded blocking-ness) for now it's ~$3 per guix build so we want to add this very sparingly :) and avoid having to push multiple times with it :)

I tried to get a pricing for an AWS instance that is running 100% time (shared instance). Benefits of this solution, that we can preserve a directory `/gnu` between the builds easily that is an actually cache for builds. With using cache the builds are 10 times faster and don't require 48 cpus with 64Gb ram to see result fast. Except case when there's major changes such as new version of `guix.
The instance that I tried to get estimation of cost is:
 - 8 vCPU
 - 16Gb RAM
 - storage/network
If it is utilized 100% time (builds-follow-new-build-again and again) it is claimed as cost 250$/month, if utilization is 15% only CPU - than 25$/month.

@strophy Can we consider instead of creating many-core-cpu instance to build everything from scratch that is VERY CPU greedy --- to use a constantly running shared server with much tinier hardware but 24/7?
It's easy to implement cache (that will preserve more than 90% resources) for build in this case?

File: .github/workflows/guix-build.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+      - name: Run Guix build
+        run: |
+          export ADDITIONAL_GUIX_COMMON_FLAGS='--max-jobs=32' && \
+          mkdir -p depends/SDKs && \
+          mkdir -p /tmp/guix-store && \
+          curl -L https://bitcoincore.org/depends-sources/sdks/Xcode-12.1-12A7403-extracted-SDK-with-libcxx-headers.tar.gz | tar -xz -C depends/SDKs && \
+          docker run --privileged -d --name guix-daemon --rm -v ${{ github.workspace }}:/dash -w /dash alpine-guix:latest && \
+          docker exec guix-daemon bash -c "\
+          chmod 777 /dash/depends && \
```
Comment (UdjinM6) [inline .github/workflows/guix-build.yml:49]: why 777?

Comment (PastaPastaPasta) [reply .github/workflows/guix-build.yml:49]: I was getting some weird permissions issues and it's in the docker container / action so I figured it's not an issue to just fully open up the permissions.

**Review by UdjinM6 — APPROVED** [review]
utACK (pls update PR title/description/milestone before merging)

**Comment by strophy** [comment]
We are already implementing S3 caching backend for platform, so most of the work is already done to support caching and ephemeral spot runners. If you guys prefer a single non-ephemeral runner that works too, we can support any config supported by https://github.com/philips-labs/terraform-aws-github-runner

**Pull request merged**